supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/request/ErrorReporter.scala

```diff
@@ -0,0 +1,98 @@
+package cromwell.backend.google.pipelines.v2alpha1.api.request
+
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import com.google.api.services.genomics.v2alpha1.model._
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation._
+import cromwell.backend.google.pipelines.common.api.RunStatus.{Cancelled, Failed, Preempted}
+import cromwell.backend.google.pipelines.v2alpha1.api.ActionBuilder.Labels.Key
+import cromwell.backend.google.pipelines.v2alpha1.api.Deserialization._
+import cromwell.backend.google.pipelines.v2alpha1.api.request.RequestHandler.logger
+import cromwell.core.{ExecutionEvent, WorkflowId}
+import io.grpc.{Status => GStatus}
+import mouse.all._
+
+import scala.collection.JavaConverters._
+
+object ErrorReporter {
+  // This can be used to log non-critical deserialization failures and not fail the task
+  implicit class ErrorOrLogger[A](val t: ErrorOr[A]) extends AnyVal {
+    private def logErrors(errors: NonEmptyList[String])(implicit workflowId: WorkflowId, operation: Operation) = {
+      logger.error(s"[$workflowId] Failed to parse PAPI response. Operation Id: ${operation.getName}" + s"${errors.toList.mkString(", ")}")
+    }
+
+    def fallBack(implicit workflowId: WorkflowId, operation: Operation): Option[A] = t match {
+      case Valid(s) => Option(s)
+      case Invalid(f) =>
+        logErrors(f)
+        None
+    }
+
+    def fallBackTo(to: A)(implicit workflowId: WorkflowId, operation: Operation): A = t match {
+      case Valid(s) => s
+      case Invalid(f) =>
+        logErrors(f)
+        to
+    }
+  }
+}
+
+class ErrorReporter(machineType: Option[String],
+                    wasPreemptible: Boolean,
+                    executionEvents: List[ExecutionEvent],
+                    zone: Option[String],
+                    instanceName: Option[String],
+                    actions: List[Action])(implicit operation: Operation, workflowId: WorkflowId) {
```

Comment:
curious why are `Operation` and `WorkflowId` implicit in all these methods & classes?  If you don't want to pass them around I'd use a `Reader` or event just a function return type

Reply:
They're there for context if we need to log an error and I didn't want to have to pass them explicitly down the stack to every function that doesn't use them. But I think you're right, `Reader` is a better way to do this, will give it a shot.